src/Npgsql/ConnectorPool.cs

```diff
@@ -197,200 +190,172 @@ internal async ValueTask<NpgsqlConnector> AllocateLong(NpgsqlConnection conn, Np
         {
             // No idle connector was found in the pool.
             // We now loop until one of three things happen:
-            // 1. The pool isn't at max capacity (Total < Max), so we can create a new physical connection.
-            // 2. The pool is at maximum capacity and there are no idle connectors (Busy == Max),
+            // 1. The pool isn't at max capacity (Open < Max), so we can create a new physical connection.
+            // 2. The pool is at maximum capacity and there are no idle connectors (Open - Idle == Max),
             // so we enqueue an open attempt into the waiting queue, so that the next release will unblock it.
-            // 3. An connector makes it into the idle list (race condition with another Release().
+            // 3. An connector makes it into the idle list (race condition with another Release()).
             while (true)
             {
                 NpgsqlConnector? connector;
-                var state = State.Copy();
-                var newState = state;
 
-                if (state.Total < _max)
+                var (openCount, idleCount, busyCount) = State;
```

Comment:
Total nit: maybe just open/idle/busy to keep variable names short?

Reply:
Yeah, so the nasty thing is we have arrays called _idle and _open as well, I don't enjoy the overloaded meanings which is why I opted to maximally disambiguate here.